packages/x-components/src/x-modules/queries-preview/components/query-preview.vue

```diff
@@ -130,16 +139,15 @@
      * @internal
      */
     @Inject()
-    public location?: FeatureLocation;
+    protected location?: FeatureLocation;
 
     /**
      * The computed request object to be used to retrieve the query preview results.
      *
      * @returns The search request object.
-     * @public
+     * @internal
      */
-    @XEmit('QueryPreviewRequestChanged', { immediate: false })
-    public get queryPreviewRequest(): SearchRequest {
+    protected get queryPreviewRequest(): SearchRequest {
       const origin = createOrigin({
         feature: this.queryFeature,
         location: this.location
```

Comment:
It is not related to this PR, but If a location is not provided by an ancestor, an error is logged in the console.
Also, should the location be a prop instead of an injected value, as well as with the queryFeature?

Reply:
Hi! Good catch! The Injection needed a default value not to warn. On the other hand, the location is managed via the  provide/inject logic (👀 LocationProvider component)